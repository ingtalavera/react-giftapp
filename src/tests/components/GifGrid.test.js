import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs")

describe('Pruebas en <GifGrid />', () => {
    const category = 'Goku';

    test('debe de cargar correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={ category }/>)

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe cargar la informacion de las imagenes useFetchGifs', () => {
        
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cosa.jpg',
                title: 'Cualquier cosa'
            },
            {
                id: 'ABD',
                url: 'https://localhost/cosa2.jpg',
                title: 'Cualquier cosa dos'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={ category }/>)
        
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    })
    
})