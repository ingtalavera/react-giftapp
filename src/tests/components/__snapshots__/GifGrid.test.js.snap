// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de cargar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
